File: Arrays & Strings/167. Two Sum II - Input Array Is Sorted.py
from typing import List

class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l, r = 0, len(numbers) - 1
        while(l < r):
            if(numbers[l] + numbers[r] < target):
                l += 1
            elif(numbers[l] + numbers[r] > target):
                r -= 1
            else:
                return [l + 1, r + 1]
        return []
        
    
    
if __name__ == "__main__":
    sol = Solution()
    numbers = [2,7,11,15]
    target = 9
    print(sol.twoSum(numbers, target))

File: Arrays & Strings/167. Two Sum II - Input Array Is Sorted.cpp
/*
167. Two Sum II - Input Array Is Sorted
*/

#include <bits/stdc++.h>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& numbers, int target) 
    {
        int l = 0, r = numbers.size() - 1;
        while (numbers[l] + numbers[r] != target)
            if (numbers[l] + numbers[r] > target)
                r--;
            else
                l++;
        return {l + 1, r + 1};
    }
};


int main() 
{
    std::vector<int> numbers = {2,7,11,15};
    int target = 9;

    Solution sol;
    std::vector<int> res = sol.twoSum(numbers, target);

    for (const int& num: res)
    {
        std::cout << num << " "; 
    }
    std::cout << std::endl;

    return 0;
}
